tests/rados/test_rados_preempt_scrub.py

```diff
@@ -0,0 +1,198 @@
+"""
+This file contains the  methods to verify the preempt messages in the OSD logs.
+"""
+
+import time
+import traceback
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.core_workflows import RadosOrchestrator
+from tests.rados.monitor_configurations import MonConfigMethods
+from utility.log import Log
+from utility.utils import method_should_succeed
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    CEPH-83572916 - Verify that the preempted messages are generated at OSD logs during scrubbing
+    1.Create pool and write data in to the pool
+    2.set the osd_shallow_scrub_chunk_max and osd_scrub_chunk_max values to 250
+    3.Start writing the data into pool and start scrub parallely
+    4.Once the scrub started check the OSD logs contain the-
+       - head preempted
+       - WaitReplicas::react(const GotReplicas&) PREEMPTED
+    5. Once the tests are complete delete the pool and remove the set values of the parameter
+    """
+
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_object = RadosOrchestrator(node=cephadm)
+    mon_obj = MonConfigMethods(rados_obj=rados_object)
+    ceph_nodes = kw.get("ceph_nodes")
+    installer = ceph_cluster.get_nodes(role="installer")[0]
+    pool_target_configs = config["verify_osd_preempt"]["configurations"]
+
+    try:
+        scrub_status_flag = False
+        # Before starting the tests check that scrubbing is not progress in the cluster
+        time_end = time.time() + 60 * 15
+        while time.time() < time_end:
+            scrub_status = check_scrub_status(rados_object)
+            if scrub_status is False:
+                scrub_status_flag = True
+                log.info("Scrubbing is not in progress.Continuing the test execution")
+                break
+            log.info("Waiting for the scrub to complete")
+            time.sleep(30)
+
+        if scrub_status_flag is False:
+            log.error(
+                "Scrubbing is in progress cannot continue the further execution of test"
+            )
+            return 1
```

Comment:
I do not feel that if cluster triggered scrubbing is happening on the cluster, that should be a reason to fail the test case.

Reply:
I included the check because, after the scrub begins, not able to tell if the user-initiated scrub or the prior scrub is responsible for the scrub state in the cluster.